apps/mymedicare_cb/models.py

```diff
@@ -51,170 +47,246 @@ def get_and_update_user(subject, mbi_hash, hicn_hash, first_name, last_name, ema
+        "crosswalk": {},
+    }
 
     try:
         # Does an existing user and crosswalk exist for SLSx username?
         user = User.objects.get(username=subject)
+        log_dict["crosswalk"] = {
+            "id": user.crosswalk.id,
+            "user_hicn_hash": user.crosswalk.user_hicn_hash,
+            "user_mbi_hash": user.crosswalk.user_mbi_hash,
+            "fhir_id": user.crosswalk.fhir_id,
+            "user_id_type": user.crosswalk.user_id_type,
+        }
 
         if user.crosswalk.user_hicn_hash != hicn_hash:
             err_msg = "Found user's hicn did not match"
-            log_get_and_update_user(auth_flow_dict, "FAIL", user, fhir_id, mbi_hash, hicn_hash, hash_lookup_type, err_msg)
+            log_dict["status"] = "FAIL"
+            log_dict["mesg"] = err_msg
+            logger.info(log_dict)
             raise BBMyMedicareCallbackCrosswalkUpdateException(err_msg)
 
         if user.crosswalk.fhir_id != fhir_id:
             err_msg = "Found user's fhir_id did not match"
-            log_get_and_update_user(auth_flow_dict, "FAIL", user, fhir_id, mbi_hash, hicn_hash, hash_lookup_type, err_msg)
+            log_dict["status"] = "FAIL"
+            log_dict["mesg"] = err_msg
+            logger.info(log_dict)
             raise BBMyMedicareCallbackCrosswalkUpdateException(err_msg)
 
         if user.crosswalk.user_mbi_hash is not None:
             if user.crosswalk.user_mbi_hash != mbi_hash:
                 err_msg = "Found user's mbi did not match"
-                log_get_and_update_user(auth_flow_dict, "FAIL", user, fhir_id, mbi_hash, hicn_hash, hash_lookup_type, err_msg)
+                log_dict["status"] = "FAIL"
+                log_dict["mesg"] = err_msg
+                logger.info(log_dict)
                 raise BBMyMedicareCallbackCrosswalkUpdateException(err_msg)
         else:
             # Previously stored value was None/Null and mbi_hash != None, update just the mbi hash.
             if mbi_hash is not None:
                 err_msg = "UPDATE mbi_hash since previous value was NULL"
-                log_get_and_update_user(auth_flow_dict, "OK", user, fhir_id, mbi_hash, hicn_hash, hash_lookup_type, err_msg)
+                log_dict["status"] = "OK"
+                log_dict["mesg"] = err_msg
+                logger.info(log_dict)
                 user.crosswalk.user_mbi_hash = mbi_hash
                 user.crosswalk.user_id_type = hash_lookup_type
                 user.crosswalk.save()
 
         # Update hash type used for lookup, if it has changed from last match.
         if user.crosswalk.user_id_type != hash_lookup_type:
-            err_msg = "UPDATE user_id_type as it has changed from the previous lookup value"
-            log_get_and_update_user(auth_flow_dict, "OK", user, fhir_id, mbi_hash, hicn_hash, hash_lookup_type, err_msg)
+            err_msg = (
+                "UPDATE user_id_type as it has changed from the previous lookup value"
+            )
```

Comment:
@nbragdon Not sure if this affects the logging output, but this got changed to a tuple?

Should be:

```suggestion
            err_msg =  "UPDATE user_id_type as it has changed from the previous lookup value"
```

Reply:
I think that is actually used to imply multiline strings - https://stackoverflow.com/questions/5437619/python-style-line-continuation-with-strings

Reply:
> I think that is actually used to imply multiline strings - https://stackoverflow.com/questions/5437619/python-style-line-continuation-with-strings

That's right. I forgot about that syntax (w/o the trailing comma). I've used this before for multiple lines, but looked odd with a single line.